pyccl/halos/halo_model.py

```diff
@@ -137,6 +140,69 @@ def profile_norm(self, cosmo, a, prof):
         norm = 1. / self._integrate_over_mf(uk0)
         return norm
 
+    def number_counts(self, cosmo, sel, na=128):
+        """ Solves the integral:
+
+        .. math::
+            nc(sel) = \\int dM\\int da\\,\\frac{dV}{dad\\Omega}\\,n(M,a)\\,sel(M,a)
+
+        where :math:`n(M,a)` is the halo mass function, and
+        :math:`sel(M,a)` is the selection function as a function of halo mass
+        and scale factor.
+
+        Note that the selection function is normalized to integrate to unity and
+        assumed to represent the selection probaility per unit scale factor and
+        per unit mass.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
+            sel (callable): function of mass and scale factor that returns the
+                possibly unnormalized selection function
```

Comment:
There should be some information on what the expected dimensions of the inputs and outputs of this function are